File: LeetCode75-sols/Two_Pointers/CPP/Max_No_of_k_Sum_Pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int count = 0;
        int left_ptr = 0;
        int right_ptr = nums.size() - 1;
        if (nums.size() == 0) {
            return 0;
        }
        while (left_ptr < right_ptr) {
            if (nums[left_ptr] + nums[right_ptr] == k) {
                left_ptr++;
                right_ptr--;
                count++;
            }
            else if (nums[left_ptr] + nums[right_ptr] < k) {
                left_ptr++;
            }
            else {
                right_ptr--;
            }
        }
        return count;
    }
};

File: LeetCode75-sols/Two_Pointers/Python/Max_No_of_k_Sum_Pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        count = 0
        left_ptr = 0
        right_ptr = len(nums) - 1
        if (len(nums) == 0):
            return 0
        while left_ptr < right_ptr:
            if (nums[left_ptr] + nums[right_ptr] == k):
                left_ptr += 1
                right_ptr -= 1
                count += 1
            elif (nums[left_ptr] + nums[right_ptr] < k):
                left_ptr += 1
            else:
                right_ptr -= 1
        return count